homeassistant/components/recorder/entity_registry.py

```diff
@@ -19,10 +17,13 @@ def async_setup(hass: HomeAssistant) -> None:
     """Set up the entity hooks."""
 
     @callback
-    def _async_entity_id_changed(event: Event) -> None:
+    def _async_entity_id_changed(
+        event: Event[er.EventEntityRegistryUpdatedData],
+    ) -> None:
         instance = get_instance(hass)
-        old_entity_id: str = event.data["old_entity_id"]
-        new_entity_id: str = event.data["entity_id"]
+        assert event.data["action"] == "update" and "old_entity_id" in event.data
```

Comment:
That is already checked in the even_filter method `entity_registry_changed_filter`. However, mypy / pyright don't know that.

Reply:
Would you please guard it with `if TYPE_CHECKING:` so we know its only for typing